AWSClientRuntime/Tests/Auth/AWSCredentialProviderTests.swift

```diff
@@ -14,14 +14,46 @@ class AWSCredentialProviderTests: XCTestCase {
     
     func testYouCanUseCustomCredentialsProvider() throws {
         let awsCredsProvider = try AWSCredentialsProvider.fromCustom(MyCustomCredentialsProvider())
-        let credentials = try awsCredsProvider.getCredentials()
+        let credentialsResult = try awsCredsProvider.getCredentials()
+        let credentials = try credentialsResult.get()
+        XCTAssertEqual(credentials.accessKey, "MYACCESSKEY")
+        XCTAssertEqual(credentials.secret, "sekrit")
+    }
+    
+    func testYouCanUseCustomCredentialsProviderFromDiffThread() throws {
+        let awsCredsProvider = try AWSCredentialsProvider.fromCustom(MyCustomThreadTestCredentialsProvider())
+        let credentialsResult = try awsCredsProvider.getCredentials()
+        let credentials = try credentialsResult.get()
         XCTAssertEqual(credentials.accessKey, "MYACCESSKEY")
         XCTAssertEqual(credentials.secret, "sekrit")
     }
 }
 
 struct MyCustomCredentialsProvider: CredentialsProvider {
-    func getCredentials() throws -> AWSCredentials {
-        return AWSCredentials(accessKey: "MYACCESSKEY", secret: "sekrit", expirationTimeout: 30)
+    func getCredentials() throws -> SdkFuture<AWSCredentials> {
+        let future = SdkFuture<AWSCredentials>()
+        future.fulfill(AWSCredentials(accessKey: "MYACCESSKEY", secret: "sekrit", expirationTimeout: 30))
+        return future
+    }
+}
+
+struct MyCustomThreadTestCredentialsProvider: CredentialsProvider {
+    func getCredentials() throws -> SdkFuture<AWSCredentials> {
+        let group = DispatchGroup()
+        let future = SdkFuture<AWSCredentials>()
+        let sleepVal = Int.random(in: 1...4)
```

Comment:
Please remove randomization in unit tests -  If we need to sleep, 100ms seems like enough.

Reply:
will do